api/base/parsers.py

```diff
@@ -42,44 +45,78 @@ def flatten_relationships(self, relationships):
         id = data.get('id')
         return {'id': id, 'target_type': target_type}
 
+    def flatten_data(self, resource_object, request_method, is_list, is_relationship):
```

Comment:
I am passing parser_context to retrieve request_method and is_relationship, but I think is_list should be an argument because I already would have called is_bulk_request(parser_context['request']) to separate out the behavior for bulk requests/single requests. I don't want to call it again.